src/fundamental_interface.jl

```diff
@@ -243,6 +243,15 @@ function gen end
 # """
 function gens end
 
+###############################################################################
+#
+#   Variable names
+#
+###############################################################################
+
+const VarName = Union{Symbol, AbstractString, Char}
+const VarNameNonSymbol = Union{AbstractString, Char}
```

Comment:
Maybe add a comment that explains why this is sometimes useful?
```suggestion

# Constructors which use `VarName` in their argument list often are implemented
# by first converting all such inputs to use `Symbol` everywhere. For disambiguation
# and to avoid infinite recursion, it then is sometimes necessary to have methods
# in pairs, one using `Symbol`, and one using "`VarName` except for `Symbol`".
# The latter is what `VarNameNonSymbol` is for.
const VarNameNonSymbol = Union{AbstractString, Char}
```